Wrote user class

justin-obrien-smoothstack opened the pull request.

justin-obrien-smoothstack committed:
Write user class

justin-obrien-smoothstack committed:
Add persistence annotations to user class

trevorhuis-smoothstack reviewed (approved):
Not much wiggle room in these entities, looks good!

trevorhuis-smoothstack commented on src/main/java/com/ss/training/utopia/counter/entity/User.java:
```diff
@@ -0,0 +1,111 @@
+	private int userId;
+
+	@Column
+	private String name, username, password, role;
```
I explicitly name the columns for each variable, but this looks like it isn't necessary. I think I developed that practice from the not wanting the same name in the POJO that was in the library database.

justin-obrien-smoothstack committed:
Change userId from int to Long

Re-generate hashCode, equals, getters/setters, UID

justin-obrien-smoothstack committed:
Mark username as unique in use class

Hopefully this will allow the user DAO to return a single user instead
of a list when finding a user by username

justin-obrien-smoothstack committed:
Add constructors to user class

justin-obrien-smoothstack committed:
Reorder constructors in user class

justin-obrien-smoothstack committed:
Merge branch 'dev' into user

justin-obrien-smoothstack committed:
Add generated value annotation to user ID

This was found to be needed when testing the user DAO

trevorhuis-smoothstack reviewed (approved):
Already reviewed.

justin-obrien-smoothstack commented:
I made some changes since then. They're pretty unremarkable though.
